src/actions/post_build.rs

```diff
@@ -47,13 +47,20 @@ pub struct PostBuildHandler {
 // is not object-safe).
 pub trait Notifier: Send {
     fn notify_begin(&self);
+    fn notify_begin_diagnostics(&self);
+    fn notify_progress(&self, update: ProgressUpdate);
+    fn notify_publish_diagnostics(&self, PublishDiagnosticsParams);
     fn notify_end(&self);
-    fn notify_publish(&self, PublishDiagnosticsParams);
+}
+
+pub enum ProgressUpdate {
+    Message(String),
+    Percentage(f64),
```

Comment:
Initially I imagined sending a message of what is being built also when we do a percentage. However it seems we can only send either a message OR a percentage. https://github.com/Microsoft/vscode-languageserver-node/pull/261#issuecomment-356769030